var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -145,12 +145,8 @@ class Cmake(Package):
     variant('openssl', default=True,  description="Enable openssl for curl bootstrapped by CMake when using +ownlibs")
     variant('ncurses', default=True,  description='Enables the build of the ncurses gui')
 
-    # Does not compile and is not covered in upstream CI (yet).
-    conflicts('%gcc platform=darwin',
-              msg='CMake does not compile with GCC on macOS yet, '
-                  'please use %apple-clang. '
-                  'See: https://gitlab.kitware.com/cmake/cmake/-/issues/21135')
-
+    # See https://gitlab.kitware.com/cmake/cmake/-/issues/21135
+    conflicts('%gcc platform=darwin', when='@:3.17')
```

Comment:
Agreed, but would keep the message:
```suggestion
    conflicts('%gcc platform=darwin', when='@:3.17',
              msg='CMake <3.18 does not compile with GCC on macOS, '
                  'please use %apple-clang or a newer CMake release. '
                  'See: https://gitlab.kitware.com/cmake/cmake/-/issues/21135')
```